engine/src/sprite-batch.cpp

```diff
@@ -293,4 +300,10 @@ void SpriteBatch::SetProjection(glm::vec2 windowSize) {
   this->windowSize = windowSize;
   this->projection =
       glm::ortho(0.0f, this->windowSize.x, this->windowSize.y, 0.0f);
-}
\ No newline at end of file
+}
+
+void SpriteBatch::SetTextureAndDimensions(GLuint texture, const int w,
+                                          const int h) {
+  this->texture = texture;
+  this->textureRect = glm::ivec4(0, 0, w, h);
+}
```

Comment:
this is a bit dirty, may just want to create a seperate rendering pipeline for text